### PR TITLE
created new folder for trend pushtopilot from Sysgain Master

### PR DESCRIPTION
**Pull request by uday-sg**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changeling
*  created a new folder for trend push to pilot solution stack

### Description of the change
@singhkay please review

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@uday-sg__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by uday-sg** [comment]
@singhkay - 
1)Travis cl is throwing following error and can you please advice on this.
Error message: 'Legal terms have not been accepted for this item on this subscription. To accept legal terms, please go to the Azure portal (http://go.microsoft.com/fwlink/?LinkId=534873) and configure programmatic deployment for the Marketplace item or create it there for the first time'\"\r\n  }\r\n}\nUser failed validation to purchase resources

2) Also we have two input parameters where user needs to enter Azure credentials for automation jobs to run. And it fails in Travis cl validation. Can you please advice how can we get around this issue

File: p2p-trend-chef-splunk/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Trend Micro - Chef - Splunk Push to Pilot
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-STARTER-TEMPLATE-with-VALIDATION%2Fazuredeploy.json" target="_blank">
```
Comment (singhkay) [inline p2p-trend-chef-splunk/README.md]: Links need to point to your template

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,1327 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "westus",
+      "allowedValues": [
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: Can these resources be deployed in other locations?

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,1327 @@
...
+        "eastus",
+        "centralus"
+      ],
+      "metadata": {
+        "description": "Location where all the resources deploy"
+      }
+    },
+    "vmAdminNameTrendDSM": {
+      "type": "string",
+      "defaultValue": "trend",
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: please remove the default value as a security best practice

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,1327 @@
...
+    "publicIPDomainNameLabelTrendDSM": {
+      "type": "string",
+      "defaultValue": "publicdnstrenddsm",
+      "metadata": {
+        "description": "TrendMicro Deep Security Manager unique public DNS prefix"
+      }
+    },
+    "managerPortTrendDSM": {
+      "type": "string",
+      "defaultValue": "443",
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: Are there other ports that the user should select?

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,1327 @@
...
+        "description": "Linux Deep Security Agent unique public DNS prefix"
+      }
+    },
+    "ubuntuOSVersion": {
+      "type": "string",
+      "defaultValue": "14.04.2-LTS",
+      "allowedValues": [
+        "12.04.5-LTS",
+        "14.04.2-LTS",
+        "15.10"
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: 15.10 image doesn't exist anymore, try 16.04.0-LTS

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,1327 @@
...
+        "description": "Count of Linux Deep Security Agents to deploy"
+      }
+    },
+    "baseUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URL for dependent deployment files",
+        "artifactsBaseUrl": ""
+      },
+      "defaultValue": "https://trendmicrop2p.blob.core.windows.net/trendmicropushtopilot"
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: These files should live in this template folder in Github

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,1327 @@
...
+      "type": "securestring",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Azure account password"
+      }
+    }
+  },
+  "variables": {
+    "location": "[parameters('location')]",
+    "location2": "East US 2",
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: What is this hardcoded location?

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,1327 @@
...
+    "TenantActivationPassword": "NA",
+    "tagvalues": {
+      "program": "p2p",
+      "project": "TrendMicro"
+    },
+    "vmAuthTypeTrendDSM": "password",
+    "vmAdminSshPublicKeyTrendDSM": "",
+    "existingSQLServerName": "",
+    "networkExists": "new",
+    "ManagerAddress": "[concat(parameters('publicIPDomainNameLabelTrendDSM'),variables('uniqueString'),'.',parameters('location'),'.cloudapp.azure.com')]",
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: camelCase

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,1327 @@
...
+    "automationAccountName": "[concat('automation',variables('uniqueString'))]",
+    "automationScriptUri": "https://raw.githubusercontent.com/sysgain/km-docker-nodejs/master/script-GetValue.ps1",
+    "automationCredentialName": "p2pazurecredential",
+    "automationVariableName": "keyValue",
+    "uniqueString": "[uniquestring(resourceGroup().id)]",
+    "keyName": "validatorkey",
+    "resourcegroupName": "[resourceGroup().name]",
+    "jobId": "3e34020c-82e6-4aaf-b092-e1a46ee74132",
+    "kmServerUrl": "[concat('http://',parameters('publicIPDomainNameLabelOrchServer'),variables('uniqueString'),'.',variables('location'),'.cloudapp.azure.com:33001')]",
+    "automationtemplatelink": "https://raw.githubusercontent.com/jefffanjoy/Test/master/Automation/GetVariableDataNestedDeploymentTemplate.json",
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: this template should be in this folder as well

File: p2p-trend-chef-splunk/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,1327 @@
...
+    "chef_node_name": "chefnodedsagent",
+    "chef_server_url": "[concat('https://',parameters('publicIPDomainNameLabelChefServer'),variables('uniqueString'),'.',parameters('location'),'.cloudapp.azure.com/organizations/',parameters('chefOrgShortName'))]",
+    "validation_client_name": "[concat(parameters('chefOrgShortName'),'-validator')]",
+    "validation_key_format": "base64encoded",
+    "runlist": "recipe[getting-started]",
+    "vmSizeOrcServer": "Standard_D1",
+    "orcServerubuntuOSVersion": "14.04.2-LTS",
+    "installOrchTemplateurl": "https://raw.githubusercontent.com/sysgain/cloudtry-p2p-public-artifacts/master/templates/TrendMicro/orchestrator-server-setup-trend.json"
```
Comment (singhkay) [inline p2p-trend-chef-splunk/azuredeploy.json]: This template should be in this folder as well

File: p2p-trend-chef-splunk/README.md (hunk 1 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/README.md]: This has been updated on new commit

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 2 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: Yes, they be deployed on other locations and user can choose the location

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 3 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: There are a lot of parameters, so just wanted to make it easy

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 4 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: User can enter different ports, but by default it is 443

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 5 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: Sure

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 7 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: We used that for automation (which can be deployed in specific locations). In the changed version we removed that

File: p2p-trend-chef-splunk/azuredeploy.json (hunk 9 above)
Comment (uday-sg) [inline p2p-trend-chef-splunk/azuredeploy.json]: Ok

**Comment by singhkay** [comment]
merged

**Pull request merged**